cpp/source/deal.II/geometry.h

```diff
@@ -72,6 +81,13 @@ class Geometry
     return _materials;
   }
 
+  void set_materials(
+      std::shared_ptr<std::unordered_map<
+          std::string, std::vector<dealii::types::material_id>>> materials)
+  {
+    _materials = materials;
+  }
+
```

Comment:
Because geometry only saves these two boundary ids. All the others are lost when the geometry is created.